### PR TITLE
GitHub Actions: Run tests for PRs

### PR DESCRIPTION
**Pull request by ExE-Boss**
This makes it so that the tests are also run for PRs.

This workflow also always uploads code coverage, even when the tests failed.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -25,16 +25,12 @@ jobs:
...
+      # Test coverage is handled by tests.yml
+      - run: npm run test
         env:
           CONTINUOUS_INTEGRATION: 1
 
-      # Upload coverage data
-      - name: Coveralls
-        uses: coverallsapp/github-action@master
-        with:
-          github-token: ${{github.token}}
```
Comment (ExE-Boss) [inline .github/workflows/publish.yml:36]: Removing this ensures that a build with passing tests, but with a failure while uploading coverage data doesn’t prevent publishing of the package.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - run: npm install
+      - run: npm run build
+      - run: npm run lint
+      - run: npm run coverage
+        env:
+          CONTINUOUS_INTEGRATION: 1
+
+      # Upload coverage data
+      - name: Coveralls
+        if: ${{ success() || failure() }}
```
Comment (devsnek) [inline .github/workflows/tests.yml:26]: this shouldn't upload if this is a push to master and the tests fail

**Comment by devsnek** [comment]
I might take a look at this and see if these two workflows can be combined. It is unfortunate to duplicate all the work of running the tests.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (ExE-Boss) [reply .github/workflows/tests.yml:26]: This only uploads coverage data, which should still be uploaded even with failing tests.

**Pull request closed without merging**